### PR TITLE
Copter: use check_failed function for ESTOP arming check

### PR DESCRIPTION
**Pull request by tatsuy**
See this #6709

### CODE REVIEW

File: ArduCopter/AP_Arming.cpp [hunk 1]
```diff
@@ -633,7 +633,7 @@ bool AP_Arming_Copter::arm_checks(AP_Arming::Method method)
         SRV_Channels::set_emergency_stop(false);
         // if we are using motor Estop switch, it must not be in Estop position
     } else if (rc().find_channel_for_option(RC_Channel::AUX_FUNC::MOTOR_ESTOP) && SRV_Channels::get_emergency_stop()){
-        gcs().send_text(MAV_SEVERITY_CRITICAL,"Arm: Motor Emergency Stopped");
+        check_failed(true, "Motor Emergency Stopped");
```
Comment (peterbarker) [inline ArduCopter/AP_Arming.cpp:636]: This should be using `display_failure`, despite what the other blocks in here think.

There's zero reason to ignore that flag - it's clear enough what the caller wants to happen.

Comment (tatsuy) [reply ArduCopter/AP_Arming.cpp:636]: If we want to use `display_failure`, we have to move these checks from arm_checks to pre_arm_checks function as like other vehicles.

Plane:
https://github.com/ArduPilot/ardupilot/blob/master/ArduPlane/AP_Arming.cpp#L103
Rover:
https://github.com/ArduPilot/ardupilot/blob/master/Rover/AP_Arming.cpp#L81

This is not a small change. Can I include it in this PR?

Comment (peterbarker) [reply ArduCopter/AP_Arming.cpp:636]: Ah.  My mistake, sorry.  I did actually look back but failed.

I'm not sure about moving these to prearm checks.  Probably worth bringing up on the devcall.

**Comment by peterbarker** [comment]
I think the behavoural change here is that if someone gets this message when they try to arm, wanders off and unsets the RC_OPTION then after this change they'll still get the error message, won't they?

It's odd either way IMO.

**Comment by tatsuy** [comment]
@peterbarker 
This change isn't affect the logic. so the behavior isn't changed.

> if someone gets this message when they try to arm, wanders off and unsets the RC_OPTION then after this change they'll still get the error message, won't they?

If RCx_OPTION is not set to ESTOP, set emergency_stop's state to false here.
https://github.com/ArduPilot/ardupilot/pull/14249/files#diff-a7eca55a970804d585a265f90ec26ce2R633

If the behavior changes, I will remove the second commit.

**Comment by peterbarker** [comment]
Merged, thanks!

**Pull request merged**